pkg/extensions/tbtc/tbtc.go

```diff
@@ -36,29 +38,73 @@ func Initialize(ctx context.Context, handle Handle) error {
 		)
 	}
 
+	err = tbtc.monitorProvideRedemptionSignature(
+		ctx,
+		exponentialBackoff,
+		105*time.Minute, // 15 minutes before the 2 hours on-chain timeout
+	)
+	if err != nil {
+		return fmt.Errorf(
+			"could not initialize provide redemption "+
+				"signature monitoring: [%v]",
+			err,
+		)
+	}
+
 	logger.Infof("tbtc extension has been initialized")
 
 	return nil
 }
 
-type depositEventHandler func(deposit string)
+type depositEvent interface {
+	getDepositAddress() string
+}
+
+type baseDepositEvent struct {
+	depositAddress string
+}
+
+func (bdd *baseDepositEvent) getDepositAddress() string {
+	return bdd.depositAddress
+}
+
+type depositRedemptionRequestedEvent struct {
+	*baseDepositEvent
+
+	digest      [32]uint8
+	blockNumber uint64
+}
```

Comment:
I am not sure about it. I see your point but as a person who did not implement this code, it is hard for me to figure out what goes where. Maybe we can update the interface of `tbtcEventsLog`  to be similar to the one in `events_tracking.go` where we have a separate struct for each type of event and then, the function adding an event to this struct can accept any parameters that are needed? We then don't need to introduce additional abstractions but just operate on simple types.

Reply:
Those abstractions are not for `tbtcEventsLog`. Maybe it's not yet clear but this is just a side use. The most important use case is that we need some kind of context in the action function. For example, look at the provide redemption signature action. Once the timeout is exceeded, we have to provide the signature but to do that we need the digest and starting block from the event which started the monitoring action. I considered several possibilites to achieve that but this one is definitely the best and safest one despite it introduces additional complexity.

Reply:
NVM, I've changed the approach and got rid of the events log and those abstractions.